kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala

```diff
@@ -273,6 +274,16 @@ class KubernetesApplicationOperation extends ApplicationOperation with Logging {
       Utils.tryLogNonFatalError(client.close())
     }
     kubernetesClients.clear()
+
+    if (expireCleanUpTriggerCacheExecutor != null) {
```

Comment:
This may need to be placed before assigning cleanupTerminatedAppInfoTrigger to null.

https://github.com/apache/kyuubi/blob/d3520ddbcea96ec55c525600126047c44c7adb35/kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala#L267

Reply:
No, we need to invalidate the cache before shutdown the executor.

Reply:
So that, the remaining terminated application can be cleaned before server stop.

Reply:
The non-null check here is not atomic, so NPE may still occur.

https://github.com/apache/kyuubi/blob/78c2a6c0ecf67e0056aef613415e113ba5a3a013/kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala#L154